Add a extra note for jdbc object store configuration (#34297)
quarkusio/quarkus: docs/src/main/asciidoc/transaction.adoc

TIP: In listener methods, you can access more information about the transaction in progress by accessing the `TransactionManager`,
which is a CDI bean and can be ``@Inject``ed.

[[jdbcstore]]

> == Configure the transaction log to be stored in a dataSource

Reviewer: Why did you call it a dataSource instead of a database. I ask because a dataSource can be backed by other storage such as a filesystem, for example, and in this case we'd recommend that users use the file or journal stores directly.

Reviewer 2: I think its because Narayana is mention in a Datasource guide section, so calling it like that creates the logical follow-up.
The usage of the database vs datasource (or even data source) is a little problem overall all our docs.

> JDBCStore is a class in the Narayana transaction management library that facilitates the storage of transaction logs in a database by using JDBC.

Reviewer: +1

> JDBCStore is our recommended approach for users needing transaction recovery capabilities, especially when running in volatile containers.

Reviewer: Where does the recommendation come from? I can understand why it might be easier to use in a cloud environment but the file or journal stores are normally recommended. Even volatile containers are normally configured to use persistent volumes.

The reason I've picked on this statement is that the JDBCStore is a lot slower than the journal or file system and  performance of transactions is one of the primary concerns with using transactions.

Reviewer 2: I worked with the original text where the recommendation was already applied. Let me draft a second version of this whole section so that I can help @zhfeng with this a bit.

Reviewer 2: @mmusgrov Whats are the benefits of this feature then, please?

Reviewer: > @mmusgrov Whats are the benefits of this feature then, please?

Yoann suggested some nice text earlier that you can use here (`Database storage of transaction logs generally performs worse than journal of filesystem storage, but may be necessary in cloud environment lacking persistent storage (for example if your application container cannot use persistent volumes).`

To enable this capability, explicitly set `quarkus.transaction-manager.object-store.type` to `jdbc`.

You can also specify the datasource name for the transaction log storage by setting `quarkus.transaction-manager.object-store.datasource`.
If no configuration is provided, Quarkus uses the xref:datasource.adoc#dev-services[default datasource].

> When the `quarkus.transaction-manager.object-store.create-table` configuration is enabled, the transaction log table gets automatically created if it does not already exist.

Reviewer 3: ```suggestion
When the `quarkus.transaction-manager.object-store.create-table` property is set to `true`, the transaction log table gets automatically created if it does not already exist.
```

> However, to enable this capability, the transaction node identifier must be set through `quarkus.transaction-manager.node-name`.

Reviewer: If the nodeId isn't set then a default value is used so this sentence is not clear - ie the transaction manager always has a nodeId (and it gets embedded in XIDs). Perhaps you are referring to requirement for a unique nodeId but if you are then the same statement will apply to any object store type and is not specific to the JDBCStore so it looks out of place to draw attention to it here.

If you enable `quarkus.transaction-manager.object-store.create-table`, the transaction log table will be created automatically if it does not exist.

> NOTE: When enabling this capability, the transaction node identifier must be set through `quarkus.transaction-manager.node-name`. Right now the datasource transaction type has to be set with `disabled` due to the known issue https://issues.redhat.com/browse/AG-209[AG-209]

Reviewer 2: ```suggestion
NOTE: When enabling this capability, the transaction node identifier must be set through `quarkus.transaction-manager.node-name`.
The recommended workaround for the known issue link:https://issues.redhat.com/browse/AG-209[AG-209] is to set the datasource transaction type to `disabled`.
```
